mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -142,49 +371,86 @@ def ensure_list(alist):  # {{{
+
+    year_offset : int, optional
+        An offset to be added to all years in the resulting array of dates.
+        This offset is typically required to convert MPAS dates
+        starting with year 0001 into dates in the range supported by xarray
+        (1678 <= year <= 2262).
+
+    Returns
+    -------
+    datetimes : list of datetime.datetime objects
+        An array that represents the xarray time coordinate for the
+        data set, based on the given MPAs time variable name.
+
+    Raises
+    ------
+    TypeError
+        If the time variable has an unsupported type (not a date string,
+        a floating-pont number of days since the start of the simulation
+        or a numpy.datatime64 object).
+
+    Authors
+    -------
     Xylar Asay-Davis
-    Last modified: 12/05/2016
+
+    Last modified
+    -------------
+    02/10/2017
     """
 
-    if isinstance(timestr, (tuple, list)):
+    if isinstance(time_variable_name, (tuple, list)):
         # we want to average the two
-        assert(len(timestr) == 2)
-        starts = get_datetimes(ds, timestr[0], yearoffset)
-        ends = get_datetimes(ds, timestr[1], yearoffset)
+        assert(len(time_variable_name) == 2)
+        starts = _get_datetimes(ds, time_variable_name[0], year_offset)
+        ends = _get_datetimes(ds, time_variable_name[1], year_offset)
         datetimes = [starts[i] + (ends[i] - starts[i])/2
                      for i in range(len(starts))]
         return datetimes
 
-    time_var = ds[timestr]
+    time_var = ds[time_variable_name]
 
     if time_var.dtype == '|S64':
         # this is a variable like date strings like 'xtime'
         time = [''.join(atime).strip() for atime in time_var.values]
-        datetimes = [datetime.datetime(yearoffset + int(x[:4]), int(x[5:7]),
-                                       int(x[8:10]), int(x[11:13]),
-                                       int(x[14:16]), int(x[17:19]))
+        datetimes = [datetime.datetime(year_offset + int(x[:4]),
+                                       int(x[5:7]), int(x[8:10]),
+                                       int(x[11:13]), int(x[14:16]),
+                                       int(x[17:19]))
                      for x in time]
     elif time_var.dtype == 'float64':
-        # this array contains floating-point days like 'daysSinceStartOfSim'
-        start = datetime.datetime(year=yearoffset+1, month=1, day=1)
+        # this array contains floating-point days like
+        # 'daysSinceStartOfSim'
+        start = datetime.datetime(year=year_offset+1, month=1, day=1)
         datetimes = [start + datetime.timedelta(x)
                      for x in time_var.values]
     elif time_var.dtype == 'timedelta64[ns]':
         # this array contains a variable like 'daysSinceStartOfSim' as a
         # timedelta64
-        start = datetime.datetime(year=yearoffset+1, month=1, day=1)
+        start = datetime.datetime(year=year_offset+1, month=1, day=1)
```

Comment:
By the way, why was `time_var` (written by you if I'm not mistaken) fine but `year_offset` was not?  Sorry.  Still venting my frustration!